app/daemon.go

```diff
@@ -114,13 +117,25 @@ func (d *MenderDaemon) Run() error {
 				*inventoryUpdateState:
 				log.Infof("Forcing state machine to: %s", nState)
 				toState = nState
+				updateLastCheckAttempt = false
 			default:
 				log.Errorf("Cannot check update or update inventory while in %s state", toState)
 			}
 
 		default:
 			// Identity op - do nothing.
 		}
+		// Set the time for the last attempts
+		switch toState.(type) {
+		case *updateCheckState:
+			if updateLastCheckAttempt {
+				d.Sctx.lastUpdateCheckAttempt = time.Now()
```

Comment:
I'm not really arguing that it's better, but rather a suggestion: Could the forced state, and the timer-expired state, be two different states? One would update the timestamp, and one would not. It's not radically different from what you are already doing, but perhaps more state-idiomatic. I dunno..

Reply:
Hmm, I kinda like the idea, but is it okay to skip it for this simple if?

Reply:
Yes, absolutely, it was just an opportunistic suggestion 🙂

Reply:
:smiley_cat: